app/views/agents/_form.html.erb

```diff
@@ -25,7 +25,7 @@
           <% if @agent.new_record? %>
             <div class="form-group type-select">
               <%= f.label :type %>
-              <%= f.select :type, options_for_select([['Select an Agent Type', 'Agent', {title: ''}]] + Agent.types.map {|type| [agent_type_to_human(type.name), type, {title: h(Agent.build_for_type(type.name,current_user,{}).html_description.lines.first.strip)}] }, @agent.type), {}, class: 'form-control', autofocus: true %>
```

Comment:
I moved `[['Select an Agent Type', 'Agent', {title: ''}]]` to the helper as well, caching everything with the `cache` helper does not seem to give significant performance improvements. So I don't think it's worth adding the JavaScript which would be needed to select the Agent type.